utils/datasets.py

```diff
@@ -0,0 +1,282 @@
+import torch
+import torchvision
+from torchvision import datasets, models, transforms
+from torch.utils.data import Dataset,DataLoader
+import torch.nn as nn
+import numpy as np
+import math
+import os
+import matplotlib.pyplot as plt
+import pandas as pd
+import logging
+import urllib
+
+# Imports
+#==================================#
+
+# Logging Configuration - Defines the format and severity of the logs 
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[
+        logging.FileHandler("datasets_debug.log"),
+        logging.StreamHandler()
+    ]
+)
+
+class Regression(Dataset):
```

Comment:
I think this class is redundant, [TensorDataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.TensorDataset) does the same thing

Reply:
did not know that I will look into it and re-implement the dataset